homeassistant/components/hue/sensor.py

```diff
@@ -0,0 +1,324 @@
+
+    @property
+    def name(self):
+        """Return a friendly name for the sensor."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return if sensor is available."""
+        return self.bridge.available and (self.bridge.allow_unreachable or
+                                          self.sensor.config['reachable'])
+
+    @property
+    def swupdatestate(self):
+        return self.sensor.raw.get('swupdate', {}).get('state')
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            'identifiers': {
+                (hue.DOMAIN, self.unique_id)
+            },
+            'name': self.name,
+            'manufacturer': self.sensor.manufacturername,
+            # productname added in Hue Bridge API 1.24
+            # (published 03/05/2018)
+            'model': self.sensor.productname or self.sensor.modelid,
+            # Not yet exposed as properties in aiohue
+            'sw_version': self.sensor.swversion,
+            'via_hub': (hue.DOMAIN, self.bridge.api.config.bridgeid),
+        }
+
+    async def async_update(self):
+        """Synchronize state with bridge."""
+        await self.async_request_bridge_update(self.sensor.id)
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {}
+        return attributes
+
+
+class GenericZLLSensor(GenericHueSensor):
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = super().device_state_attributes
+        attributes.update({
+            "battery": self.sensor.battery,
+            "last_updated": self.sensor.lastupdated,
+            "on": self.sensor.on,
+            "reachable": self.sensor.reachable,
+        })
+        return attributes
+
+
+class HueLightLevel(GenericZLLSensor, Entity):
+
+    _entity_id_format = SENSOR_ENTITY_ID_FORMAT
+    device_class = DEVICE_CLASS_ILLUMINANCE
+    unit_of_measurement = "Lux"
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.sensor.lightlevel
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = super().device_state_attributes
+        attributes.update({
+            "is_dark": self.sensor.dark,
+            "is_daylight": self.sensor.daylight,
+            "threshold_dark": self.sensor.tholddark,
+            "threshold_offset": self.sensor.tholdoffset,
+        })
+        return attributes
+
+
+class HueTemperature(GenericZLLSensor, Entity):
+
+    _entity_id_format = SENSOR_ENTITY_ID_FORMAT
+    device_class = DEVICE_CLASS_TEMPERATURE
+    unit_of_measurement = TEMP_CELSIUS
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.sensor.temperature / 100
+
+
+class HuePresence(GenericZLLSensor, BinarySensorDevice):
```

Comment:
Nope. This should be moved to a file `hue/binary_sensor.py`

Reply:
See below